pkg/apis/sources/v1alpha1/mongodbsource_types.go

```diff
@@ -59,24 +59,9 @@ var (
 	_ resourcesemantics.GenericCRD = (*MongoDbSource)(nil)
 	// Check that MongoDbSource implements the Conditions duck type.
 	_ = duck.VerifyType(&MongoDbSource{}, &duckv1.Conditions{})
-	// Check that the type conforms to the duck Knative Resource shape.
-	_ duckv1.KRShaped = (*MongoDbSource)(nil)
```

Comment:
we should conform to KRShaped. You will have to implement two methods for that. 
Look at the examples in knative-gcp sources. And you should add two simple UTs as well..
https://github.com/google/knative-gcp/blob/5bdd3d75c6f93107681a0ad4319dd8278708bdd7/pkg/apis/events/v1beta1/cloudauditlogssource_types.go#L132
https://github.com/google/knative-gcp/blob/5bdd3d75c6f93107681a0ad4319dd8278708bdd7/pkg/apis/events/v1beta1/cloudauditlogssource_types.go#L138

https://github.com/google/knative-gcp/blob/master/pkg/apis/events/v1beta1/cloudbuildsource_types_test.go

Reply:
done, thx